WordPress/wordpress-playground
# Add `/wp-content/fonts/` directory to file paths

## What is this PR doing?
Adds support for the Font Library's font storage location, `/wp-content/fonts/` as of https://github.com/WordPress/gutenberg/pull/54122.

## What problem is it solving?
Fonts uploaded into a Playgrounds instance do not have their paths resolved properly, resulting in a 404 for user-uploaded font files.

## How is the problem addressed?
Adds `/wp-content/fonts/` to the allowed file paths, so that user-uploaded font paths are rewritten for browser in-memory scope.

## Testing Instructions
Run/compare the following Playgrounds instances, which include [Gutenberg PR #53884](https://github.com/WordPress/gutenberg/pull/53884):
- [Playground (production)](https://playground.wordpress.net/#{"landingPage":%20"/wp-admin/site-editor.php?canvas=edit",%20"steps":%20[{"step":%20"defineWpConfigConsts",%20"consts":%20{"FONT_LIBRARY_ENABLE":%20true,%20"WP_DEVELOPMENT_MODE":%20"all"%20}%20},%20{"step":%20"login",%20"username":%20"admin",%20"password":%20"password"},%20{"step":%20"mkdir",%20"path":%20"/wordpress/pr"},%20{"step":%20"writeFile",%20"path":%20"/wordpress/pr/pr.zip",%20"data":%20{"resource":%20"url",%20"url":%20"/plugin-proxy?org=WordPress&repo=gutenberg&workflow=Build%20Gutenberg%20Plugin%20Zip&artifact=gutenberg-plugin&pr=53884",%20"caption":%20"Downloading%20Gutenberg%20PR%2053884"},%20"progress":%20{"weight":%202,%20"caption":%20"Applying%20Gutenberg%20PR%2053884"}%20},%20{"step":%20"unzip",%20"zipPath":%20"/wordpress/pr/pr.zip",%20"extractToPath":%20"/wordpress/pr"},%20{"step":%20"installPlugin",%20"pluginZipFile":%20{"resource":%20"vfs",%20"path":%20"/wordpress/pr/gutenberg.zip"}%20}]%20})
- [Playground (local with patch)](http://localhost:5400/website-server/#{%22landingPage%22:%20%22/wp-admin/site-editor.php?canvas=edit%22,%20%22steps%22:%20[{%22step%22:%20%22defineWpConfigConsts%22,%20%22consts%22:%20{%22FONT_LIBRARY_ENABLE%22:%20true,%20%22WP_DEVELOPMENT_MODE%22:%20%22all%22%20}%20},%20{%22step%22:%20%22login%22,%20%22username%22:%20%22admin%22,%20%22password%22:%20%22password%22},%20{%22step%22:%20%22mkdir%22,%20%22path%22:%20%22/wordpress/pr%22},%20{%22step%22:%20%22writeFile%22,%20%22path%22:%20%22/wordpress/pr/pr.zip%22,%20%22data%22:%20{%22resource%22:%20%22url%22,%20%22url%22:%20%22/plugin-proxy?org=WordPress&repo=gutenberg&workflow=Build%20Gutenberg%20Plugin%20Zip&artifact=gutenberg-plugin&pr=53884%22,%20%22caption%22:%20%22Downloading%20Gutenberg%20PR%2053884%22},%20%22progress%22:%20{%22weight%22:%202,%20%22caption%22:%20%22Applying%20Gutenberg%20PR%2053884%22}%20},%20{%22step%22:%20%22unzip%22,%20%22zipPath%22:%20%22/wordpress/pr/pr.zip%22,%20%22extractToPath%22:%20%22/wordpress/pr%22},%20{%22step%22:%20%22installPlugin%22,%20%22pluginZipFile%22:%20{%22resource%22:%20%22vfs%22,%20%22path%22:%20%22/wordpress/pr/gutenberg.zip%22}%20}]%20})

1. Open *Styles > Typography*.
2. Activate the Font management dialog by clicking the "Aa" icon above the fonts list.
3. Upload a font file (e.g. `.woff2` or `.ttf` file). The font should activate automatically.
4. Select the Headings element and assign the uploaded font from the picker.
5. Click **Activate & Save**.
6. Navigate to the frontend of the Playgrounds site.
7. Observe the font applied to headings. The font file should load properly (no 404).
8. Click "Edit Site" to navigate back to the editor.
9. Observe the font applied to headings in the editor. It should load properly.

### Note Regarding Initial Font Display in Editor
In Step 4, the new font may not display correctly on headings. This is due to an issue that will be addressed in https://github.com/WordPress/gutenberg/pull/54313. Until this is merged, the editor will need to be reloaded in order to account for the newly added font file.
